### PR TITLE
Add pagination to Back list products URL

### PR DESCRIPTION
**Pull request by wdammak**

### CODE REVIEW

File: htdocs/product/list.php [hunk 1]
```diff
@@ -711,7 +711,8 @@
 		if (! empty($arrayfields['p.ref']['checked']))
 		{
 			print '<td class="tdoverflowmax200">';
-			print $product_static->getNomUrl(1);
+			$pagination = 'page='.$page.'&limit='.$limit.'&contextpage='.$contextpage.'&type='.$type.'&search_type='.$search_type.'&sortfield='.$sortfield.'&sortorder='.$sortorder;
+			print $product_static->getNomUrl(1,'',0,-1,$pagination);
```
Comment (eldy) [inline htdocs/product/list.php:716]: What is goal of this ? Fix a bug ? Can you describe how to reproduce ?

Comment (wdammak) [reply htdocs/product/list.php:716]: https://github.com/Dolibarr/dolibarr/issues/9625

Comment (wdammak) [reply htdocs/product/list.php:716]: these parameters will be added to the "return list" link to return to the starting point and not to the zero point as is currently the case.

**Comment by wdammak** [comment]
Do you have another solution?

**Comment by eldy** [comment]
There is currently a feature called "save_lastsearch_values / restore_lastsearch_values" that exists. May be we can use it. To make a more complete answer, can you describe how you reproduce the trouble on a develop version ?

**Comment by wdammak** [comment]
- Go to the list of products: http://localhost/develop/product/list.php?type=0
- Go to page 2 of the list of products with pagination: http://localhost/develop/product/list.php?page=2&contextpage=productlist&type=0&search_type=0&sortfield=p.ref&sortorder=ASC
- Choose one of the products from this list on page 2 to see its profile: http://localhost/develop/product/card.php?id=64&save_lastsearch_values=1
- Then you click on the URL **Back List**. 

**Bug:** This url instead of you refers to page 2 of the list, it refers you to the first page and that's the bug.

**It must return me to the starting page (page 2) and not the start page (page 1)**

The url **Back List** should refer me to this url: http://localhost/develop/product/list.php?page=2&contextpage=productlist&type=0&search_type=0&sortfield=p.ref&sortorder=ASC

And **NOT** the current url: http://localhost/develop/product/list.php?restore_lastsearch_values=1&type=0

I hope it's pretty clear :)

**Comment by wdammak** [comment]
https://github.com/Dolibarr/dolibarr/issues/9625#issuecomment-429055630
il m'a compris rapidement :)

**Comment by wdammak** [comment]
Otherwise with cookies to keep the simple links

**Pull request closed without merging**